lms/djangoapps/grades/models.py

```diff
@@ -336,3 +340,59 @@ def update(
         self.possible_graded = possible_graded
         self.visible_blocks_id = visible_blocks_hash  # pylint: disable=attribute-defined-outside-init
         self.save()
+
+
+class SubsectionGradesEnabledFlag(ConfigurationModel):
+    """
+    Enables subsection grades across the platform.
+    When this feature flag is set to true, individual courses
+    must also have subsection grades enabled for the
+    feature to take effect.
+    """
+
+    @classmethod
+    def feature_enabled(cls, course_id=None):
+        """
+        Looks at the currently active configuration model to determine whether the subsection grades feature is
+        available.
+
+        If the flag is not enabled, the feature is not available.
+        If the flag is enabled and the provided course_id is authorized,
+            the feature is available.
+        If the flag is enabled and course-specific authorization is not required, the feature is available.
+        """
+        if not SubsectionGradesEnabledFlag.is_enabled():
+            return False
+        elif course_id:
+            return CourseAuthorization.subsection_grades_enabled(course_id)
+        return True
```

Comment:
I'm not sure I like the logic of always assuming `True` if the global flag is on and no course id is provided. Could we try something similar to bulk email, where the global flag has both an enabled boolean, and a "require individual courses to be authorized" boolean?